src/NUnitFramework/framework/Internal/Results/TestSuiteResult.cs

```diff
@@ -112,48 +132,59 @@ public override IList<ITestResult> Children
         /// <param name="result">The result to be added</param>
         public virtual void AddResult(ITestResult result)
         {
-            Children.Add(result);
+#if PARALLEL
+            _children.Enqueue(result);
+#else
+            _children.Add(result);
+#endif
 
```

Comment:
I fixed the issue.  However, since since Children can be overriden in a derived class (as is the case in TestCaseResult), Children for the current instance may be any IEnumerable<ITestResult>.